app/main.py

```diff
@@ -1,6 +1,16 @@
 from typing import Callable
+_cache = {}
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    def wrapper(*args) -> int:
+        if func not in _cache:
+            _cache[func] = {}
+        if args not in _cache[func]:
```

Comment:
1. Just check if args not in your cache bucket. 
2. If that args not in your cache storage , get the func(*args) as the result and then add as a key for your sorage this way : CACHE_STORAGE[args] = result

Reply:
I don't understand how my decorator can work with different functions. I have a global variable where i store common cache for some functions